test/test_validation.py

```diff
@@ -129,19 +129,22 @@ def test_section_unique_ids(self):
 
     def test_standalone_section(self):
         """
-        Test if standalone section is validated right.
+        Test if standalone section does not return errors.
         """
 
         sec_one = odml.Section("sec1")
-        sec_two = odml.Section("sec2", parent=sec_one)
-        prop = odml.Property("prop", parent=sec_one, values=[1, 2])
+        odml.Section("sec2", parent=sec_one)
+        odml.Property("prop", parent=sec_one, values=[1, 2])
 
-        validate(sec_one)
+        for err in validate(sec_one).errors:
+            assert("error" not in err.rank)
 
     def test_standalone_property(self):
         """
-        Test if standalone property is validated right.
+        Test if standalone property does not return errors.
         """
 
         prop = odml.Property("prop", dtype="int", values=[1, 2])
-        validate(prop)
+
+        for err in validate(prop).errors:
+            assert("error" not in err.rank)
```

Comment:
Some tests with invalid objects that check if the correct errors are produced should also be added.

More useful would be to test the new functionality.  This PR introduces validation on object creation, so the new functionality should be tested: Are the appropriate error messages printed when an object with issues is created?  Since we just print the errors to `stdout` when this happens and the errors aren't stored anywhere after that, it might be necessary to capture `stdout` in the tests and check if the expected output was printed.